scripts/check_runtime

```diff
@@ -0,0 +1,294 @@
+    end
+
+    scope_name || "Object"
+  end
+
+  class Context
+    TEST_NAME = "test.rb"
+
+    def initialize(gem_name)
+      @gem_name = gem_name
+      @gemfile = String.new
+      @requires = String.new
+      @body = String.new
+      @workdir = Dir.mktmpdir
+    end
+
+    def run!
+      write_gemfile!
+      write_test!
+
+      out, status = bundle_install!
+      unless status.success?
+        $stderr.puts(" * Can't install gem `#{@gem_name}` (#{out.gsub(/\n/, "")})")
+        return false
+      end
+
+      out, status = run_test!
+      unless status.success?
+        $stderr.puts(out)
+        return false
+      end
+
+      true
+    ensure
+      FileUtils.rm_rf(@workdir)
+    end
+
+    def add_gem_dependency(name)
+      @gemfile << <<~RB
+        gem "#{name}"
+      RB
+    end
+
+    def add_require(name)
+      @requires << <<~RB
+        begin
+          require "#{name}"
+        rescue LoadError => e
+          $stderr.puts(" * Can't require `#{name}`")
+          $success = false
+        end
+      RB
+    end
+
+    def add_constant(const_name, loc)
+      @body << <<~RB
+        __rbi_repo_get_const("#{const_name}", "#{loc.to_s}")
+      RB
+    end
+
+    def add_method(recv_name, method_name, loc, allow_missing:, singleton: false)
+      @body << <<~RB
+        __rbi_repo_get_method(
+          "#{recv_name}",
+          "#{method_name}",
+          "#{loc.to_s}",
+          singleton: #{singleton},
+          allow_missing: #{allow_missing}
+        )
+      RB
+    end
+
+    private
+
+    def write_gemfile!
+      File.write("#{@workdir}/Gemfile", <<~GEMFILE)
+        source "https://rubygems.org"
+        #{@gemfile}
+        gem "#{@gem_name}"
+      GEMFILE
+    end
+
+    def write_test!
+      File.write("#{@workdir}/#{TEST_NAME}", ruby_string)
+    end
+
+    def bundle_install!
+      Bundler.with_unbundled_env do
+        Open3.capture2e("bundle config set --local path 'vendor/bundle'", chdir: @workdir)
+        Open3.capture2e("bundle install --quiet", chdir: @workdir)
+      end
+    end
+
+    def run_test!
+      Bundler.with_unbundled_env do
+        Open3.capture2e("bundle exec ruby #{TEST_NAME}", chdir: @workdir)
+      end
+    end
+
+    def ruby_string
+      <<~RB
+        $success = true
+
+        def __rbi_repo_get_const(const_name, rbi_loc)
+          Kernel.const_get("\#{const_name}")
+        rescue NameError => e
+          $stderr.puts(" * Missing runtime constant `\#{const_name}` (defined at \#{rbi_loc})")
+          $success = false
+          nil
+        end
+
+        def __rbi_repo_get_method(recv_name, method_name, rbi_loc, singleton:, allow_missing:)
+          const = __rbi_repo_get_const(recv_name, rbi_loc)
+          return unless const
+
+          if singleton
+            const.method("\#{method_name}")
+          else
+            const.instance_method("\#{method_name}")
+          end
+        rescue NameError => e
```

Comment:
We could use [method_defined?](https://ruby-doc.org/core-3.1.2/Module.html#method_defined-3F-method) on `const` and `const.singleton_class` so that we don't use exceptions as control flow.

Reply:
Since I have to implement the `rescue` for both `__rbi_repo_get_const` and `__rbi_repo_respond_to_method_missing` it made more sense to keep it here as well.